src/interpreter/executors/main.rs

```diff
@@ -27,171 +45,40 @@ impl Interpreter<PredicateStorage> {
     ///
     /// This is not a valid entrypoint for debug calls. It will only return a `bool`, and not the
     /// VM state required to trace the execution steps.
-    pub fn check_predicates(tx: CheckedTransaction, params: ConsensusParameters) -> bool {
+    pub fn check_predicates<S, Tx>(checked: Checked<Tx, S>, params: ConsensusParameters) -> bool
+    where
+        S: Stage,
+        Tx: ExecutableTransaction,
+        <Tx as IntoChecked>::Metadata: CheckedMetadata,
+    {
         let mut vm = Interpreter::with_storage(PredicateStorage::default(), params);
 
-        if !tx.as_ref().check_predicate_owners() {
+        if !checked.transaction().check_predicate_owners() {
             return false;
         }
 
-        let predicates: Vec<RuntimePredicate> = tx
-            .as_ref()
+        let predicates: Vec<RuntimePredicate> = checked
+            .transaction()
             .inputs()
             .iter()
             .enumerate()
-            .filter_map(|(idx, _)| RuntimePredicate::from_tx(&params, tx.as_ref(), idx))
+            .filter_map(|(idx, _)| RuntimePredicate::from_tx(&params, checked.transaction(), idx))
             .collect();
 
         predicates
             .into_iter()
-            .fold(vm.init_predicate(tx), |result, predicate| -> bool {
+            .fold(vm.init_predicate(checked), |result, predicate| -> bool {
                 // VM is cloned because the state should be reset for every predicate verification
                 result && vm.clone()._check_predicate(predicate)
             })
     }
-
-    /// Initialize the VM with the provided transaction and check the input predicate indexed by
-    /// `idx`. If the input isn't of type [`Input::CoinPredicate`], the function will return
-    /// `false`.
-    ///
-    /// For additional information, check [`Self::check_predicates`]
-    pub fn check_predicate(&mut self, tx: CheckedTransaction, idx: usize) -> bool {
```

Comment:
It is not used in the `fuel-vm` or `fuel-core`, so I removed it

Reply:
Eventually we want to be able to parallelize predicate verification which is why the option to do it on a per input basis was created, but we can add this back later.